canopen/objectdictionary/__init__.py

```diff
@@ -384,10 +391,13 @@ def add_bit_definition(self, name: str, bits: List[int]) -> None:
 
     def decode_raw(self, data: bytes) -> Union[int, float, str, bytes, bytearray]:
         if self.data_type == VISIBLE_STRING:
-            return data.rstrip(b"\x00").decode("ascii", errors="ignore")
+            # Strip any trailing NUL characters from C-based systems
+            return data.decode("ascii", errors="ignore").rstrip("\x00")
         elif self.data_type == UNICODE_STRING:
-            # Is this correct?
-            return data.rstrip(b"\x00").decode("utf_16_le", errors="ignore")
+            # The canopen standard does not specify the encoding. This
+            # library assumes utf-16-le, being the most common encoding format.
```

Comment:
"UTF-16" is ambiguous without byte order. When using IANA naming, it should be "UTF-16LE".

Reply:
Well, that's what I've been saying, the LE part is strictly defined by CANopen, because `ARRAY OF UNSIGNED16` means each glyph is represented as a 16-bit number in little-endian byte order.  What is not defined by the standard is the encoding (mapping of glyphs to numbers), so the comment should make clear that this little bit is the library's design choice.

Reply:
Yes, I think we agree, so maybe we talk past each other. I simply meant to change line 398 from "UTF-16" to "UTF-16LE". Regardless of any chose encoding extensions, reading the stream as big-endian will fail miserably, so it must be specified as LE.

Reply:
Of course the code needs to specify utf-16-le, as it does.  I was just pointing out that the little-endian part is NOT a design choice, but mandated by the standard.  Therefore IMO the comment should be restricted to the encoding part, to make it clear that UTF-16 is *chosen*, leading to utf-16-le combined with the *specification*.  It was a deliberate choice to leave out the LE part when talking about the design choice.